VigilanceMeteoFrance/VigilanceMeteoFrance/Program.cs

```diff
@@ -0,0 +1,157 @@
+            Vigilance Test = new Vigilance();
+            string risque = "";
+            if (Departement == "92" || Departement == "93" || Departement == "94")
+            {
+                Departement = "75";
+            }
+            var doc = XDocument.Load(url);
+            var depnode = doc.Descendants("risque").Where(c => c.Parent.Attribute("dep").Value == Departement);
+            if (depnode.Any())
+            {
+                List<Type> Type = new List<Type>();
+                foreach (var x in depnode)
+                {
+                    switch (x.Attribute("val").Value)
+                    {
+                        case "1":
+                            risque = "vent";
+                            break;
+                        case "2":
+                            risque = "pluie-inondation";
+                            break;
+                        case "3":
+                            risque = "orages";
+                            break;
+                        case "4":
+                            risque = "inondations";
+                            break;
+                        case "5":
+                            risque = "neige-verglas";
+                            break;
+                        case "6":
+                            risque = "canicule";
+                            break;
+                        case "7":
+                            risque = "grand-froid";
+                            break;
+                        case "8":
+                            risque = "avalanches";
+                            break;
+                        case "9":
+                            risque = "vagues-submersion";
+                            break;
+                        default:
+                            risque = "aucun";
+                            break;
+                    }
+                    Type.Add(new Type() { Name = risque });
+                    Test.Level = Int32.Parse(x.Parent.Attribute("coul").Value);
+                    PackageHost.WriteInfo("Vigilance {0} niveau {1} pour le département {2}", risque, Test.Level, Departement);
+                }
+                Test.Type = Type;
+
+                return Test;
+            }
+            else
+            {
+                var bookNodes = doc.Descendants("DV").Where(c => c.Attribute("dep").Value == Departement);
+                List<Type> Type = new List<Type>();
+                foreach (var x in bookNodes)
+                {
+                    Test.Level = Int32.Parse(x.Attribute("coul").Value);
+                }
+                Test.Type = null;
+                PackageHost.WriteInfo("Vigilance niveau {0} pour le département {1}", Test.Level, Departement);
+                return Test;
+            }
+        }
+
+        /// <summary>
+        /// Vigilance.
+        /// </summary>
+        [StateObject]
+        public class Vigilance
```

Comment:
You class must contain a constructor.
You must initialize all of your variable, in constructor to avoid, K.O limits and crash. (Sample : this.Type = new List<Type>(), avoid a crash if your list isn't instanciated).